moshi/src/main/java/com/squareup/moshi/JsonUtf8Writer.java

```diff
@@ -244,6 +245,24 @@ private void writeDeferredName() throws IOException {
     return this;
   }
 
+  @Override public JsonWriter value(BufferedSource source) throws IOException {
+    if (promoteValueToName) {
```

Comment:
I don’t think we should support this.

Reply:
... instead we should probably just fail. It’s weird.

Reply:
Agree! Hurts my head to think about this usage. IllegalStateException is best.